flocker/control/_persistence.py

```diff
@@ -143,6 +143,21 @@ def upgrade_from_v2(cls, config):
         }
         return dumps(decoded_config)
 
+    @classmethod
+    def upgrade_from_v3(cls, config):
+        """
+        Migrate a v3 JSON configuration to v4.
+
+        :param bytes config: The v3 JSON data.
+        :return bytes: The v4 JSON data.
+        """
+        decoded_config = loads(config)
+        decoded_config[u"version"] = 4
+        decoded_config[u"deployment"][u"persistent_state"] = {
```

Comment:
:grey_question: confusing to have something named u"persistent_state" in a file named "_persistence" suggests that everything in this file could be under this field.

For instance, why don't leases belong within this structure?